Composer/packages/form-dialogs/src/atoms/utils.ts

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
I see some stuff here that doesn't look specific to form dialogs. i'd like to:

1. document the util functions a bit more
2. have a conversation to understand if any of these should be pulled in from a higher scope

Reply:
3. It would be nice to have some unit tests around the utility functions that non-trivial (are doing indexing math into arrays).

Reply:
Other than the function giving the first available index for clone naming, everything else is specific to form dialog json schema. we can definitely move the other function up if there are other usecases.